ext/standard/string.c

```diff
@@ -1693,6 +1693,11 @@ PHPAPI char *php_stristr(char *s, char *t, size_t s_len, size_t t_len)
 }
 /* }}} */
 
+PHPAPI const char *php_stristr_no_tolower(const char *s, const char *t, size_t s_len, size_t t_len)
+{
+	return (const char*)php_memnistr(s, t, t_len, s + s_len);
```

Comment:
```suggestion
	if (t_len > s_len) {
		return NULL; /* Needle longer than haystack */
	}
	return (const char*)php_memnistr(s, t, t_len, s + s_len);
```

It seems like PHP_FUNCTION(stripos) asserts this precondition, but the function being exposed to other PECLs/modules for future use doesn't. The implementation fetches `haystack[needle_len - 1]` so this should be checked. Maybe add a ZEND_ASSERT too?

Reply:
I think we're technically fine, as we only access `p[needle_len - 1]` after `p <= end` where `end -= needle_len`. But an early assertion could make this clearer.

Reply:
```
end -= needle_len;
```

I'd missed that part.

Still, php strings have no length limit (e.g. needle_len can even be larger than 4GB, so I think it might be possible that `end - needle_len` underflows, even with smaller needle_len values, so the check should still be there?

Reply:
You're right. I'll add the check.